maas/resource_maas_boot_resource.go

```diff
@@ -0,0 +1,174 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMAASBootResources() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage MAAS bootresources.",
+		CreateContext: resourceBootResourcesCreate,
+		ReadContext:   resourceBootResourcesRead,
+		UpdateContext: resourceBootResourcesUpdate,
+		DeleteContext: resourceBootResourcesDelete,
+
+		Schema: map[string]*schema.Schema{
+			"boot_source": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The boot source database ID this resource is associated with.",
+			},
+			"boot_source_selections": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "The set of database IDs for boot source selections to attach to this boot resource",
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+		},
+	}
+}
+
+func resourceBootResourcesCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ClientConfig).Client
+
+	for isImporting(client) {
+		time.Sleep(5 * time.Second)
```

Comment:
let's convert this and any other sleep to `retry.StateChangeConf` and `stateConf.WaitForStateContext`. We already have an example at the `func waitForMachineStatus`